lib/cfg-lexer.c

```diff
@@ -313,6 +314,29 @@ cfg_lexer_map_word_to_token(CfgLexer *self, CFG_STYPE *yylval, const CFG_LTYPE *
   return tok;
 }
 
+int
+cfg_lexer_map_filterx_word_to_token(CfgLexer *self, CFG_STYPE *yylval, const CFG_LTYPE *yylloc, const char *token)
+{
+  int tok = cfg_lexer_map_word_to_token(self, yylval, yylloc, token);
+
+  if (tok == LL_IDENTIFIER)
+    {
+      /*
+       * Regular functions and generator functions have the exact same syntax,
+       * but we need to have different tokens for them as they are handled in
+       * completely different grammar rules.
+       */
+
+      PluginContext *plugin_context = &self->cfg->plugin_context;
+
+      if ((self->cfg && plugin_is_plugin_available(plugin_context, LL_CONTEXT_FILTERX_FUNC, token)) ||
+          (self->cfg && plugin_is_plugin_available(plugin_context, LL_CONTEXT_FILTERX_SIMPLE_FUNC, token)) ||
```

Comment:
this would not be needed at all, if

1) we had a single context for all kinds of functions (why do we need separate ones?)
2) we pushed the right context from the grammar
3) builtin functions would also be registered as plugins

then we would have a single context based plugin lookup. 

I might be missing something though, so let's discuss.

Reply:
Probably the bottleneck is my understanding of the lexer/parser. I did not find the right spot to push the context, but it might be trivial. Sure, let's talk about this IRL, maybe there is a cleaner solution.

Reply:
Fixed in the way we have decided IRL.